Dashboard/extract.py

```diff
@@ -0,0 +1,119 @@
+"Extract Script: Pulls product information from ASOS webspage"
+import json
+import logging
+import requests
+from bs4 import BeautifulSoup
+
+DEFAULT_TIMEOUT = 30
+
+
+def get_product_page(url: str, headers: dict) -> str | None:
+    """Fetch the HTML content of a product page from a given URL."""
+    if not url:
+        logging.error("URL is empty")
+        return None
+    try:
+        response = requests.get(url, headers=headers, timeout=DEFAULT_TIMEOUT)
```

Comment:
If you take my earlier recommendation, please change this to the new name:
```suggestion
        response = requests.get(url, headers=headers, timeout=DEFAULT_REQUEST_TIMEOUT_SECONDS)
```